Sources/SotoCore/Credential/ConfigFileCredentialProvider.swift

```diff
@@ -94,62 +88,11 @@ class AWSConfigFileCredentialProvider: CredentialProviderSelector {
     }
 
     static func sharedCredentials(from byteBuffer: ByteBuffer, for profile: String) throws -> StaticCredential {
-        let string = byteBuffer.getString(at: 0, length: byteBuffer.readableBytes)!
-        var parser: INIParser
-        do {
-            parser = try INIParser(string)
-        } catch INIParser.Error.invalidSyntax {
-            throw ConfigFileError.invalidCredentialFileSyntax
-        }
-
-        guard let config = parser.sections[profile] else {
-            throw ConfigFileError.missingProfile(profile)
-        }
-
-        guard let accessKeyId = config["aws_access_key_id"] else {
-            throw ConfigFileError.missingAccessKeyId
-        }
-
-        guard let secretAccessKey = config["aws_secret_access_key"] else {
-            throw ConfigFileError.missingSecretAccessKey
-        }
+        let credentials = try ConfigFileLoader.loadCredentials(from: byteBuffer, for: profile, sourceProfile: nil)
 
-        let sessionToken = config["aws_session_token"]
-
-        return StaticCredential(accessKeyId: accessKeyId, secretAccessKey: secretAccessKey, sessionToken: sessionToken)
+        return StaticCredential(accessKeyId: credentials.accessKey,
+                                secretAccessKey: credentials.secretAccessKey,
+                                sessionToken: credentials.sessionToken)
```

Comment:
If you construct either a `StaticCredential` or a `STSAssumeRoleCredentialProvider` here everything should work